cookbooks/bcpc/attributes/default.rb

```diff
@@ -24,6 +24,37 @@
 default['bcpc']['bootstrap']['preseed']['late_command'] = "true"
 default['bcpc']['bootstrap']['admin_users'] = []
 
+#
+# The node_number is used to derive Kafka broker IDs, Zookeeper myid
+# files, keepalived node priorities, and other values.  It must be
+# unique within a cluster.
+#
+# The node number is generated from the integer value of the
+# default interface mac_address, modulo Java's Integer.MAX_VALUE.
+#
+# On the provisioning node and during early bootstrap, we won't have
+# any of these values, in which case we just don't set the
+# node_number.
+#
+default_interface = begin
+                      interface_name = node[:network][:default_interface]
+                      node[:network][:interfaces][interface_name]
+                    rescue
+                      nil
```

Comment:
Should we not `raise`? Originally `chef-bcpc` would try its darnedest to never fail -- this would result in swiss cheese clusters which did not have key attributes set and debugging why a cluster was not stable or did not come up at all was quite difficult.